systemd_ctypes/bus.py

```diff
@@ -331,162 +332,491 @@ def add_object(self, path, obj):
+        return False and other handlers will have a chance to run.  If no
+        handler handles the message, systemd will generate a suitable error
+        message and send that, instead.
+
+        :message: the message that was received
+        :returns: True if the message was handled
+        """
         raise NotImplementedError
 
+
 class Object(BaseObject):
-    @classmethod
-    def prepare(cls, iface):
-        cls._dbus_interface = iface
-        cls._dbus_members = {'properties': {}, 'methods': {}, 'signals': {}}
-        for key, value in cls.__dict__.items():
-            if not key.startswith('_') and hasattr(value, '_dbus_info'):
-                category, member, info = value._dbus_info
-                cls._dbus_members[category][member] = value, info
-        cls._dbus_xml = introspection.to_xml({
-            cls._dbus_interface: {
-                'methods': {
-                    name: {
-                        'in': list(in_args),
-                        'out': list(out_args),
-                    } for name, (_, (out_args, in_args)) in cls._dbus_members['methods'].items()
-                },
-                'properties': {
-                    name: {
-                        'flags': 'r',
-                        'type': dbus_type,
-                    } for name, (_, (dbus_type,)) in cls._dbus_members['properties'].items()
-                },
-                'signals': {
-                    name: {
-                        'in': list(in_args),
-                    } for name, (_, (in_args,)) in cls._dbus_members['signals'].items()
-                },
-            }
-        })
-        return cls
+    """The high-level base class for exporting objects on D-Bus
 
-    @staticmethod
-    def _signal_wrapper(obj, member, in_types):
-        def invoker(self, *args):
-            obj(self, *args)
-            message = self._bus.message_new_signal(self._path, self._dbus_interface, member,
-                                                   in_types, *args)
-            self._bus.send(message, None)
-        return invoker
+    This class provides high-level APIs for defining methods, properties, and
+    signals, as well as implementing introspection.
 
-    @staticmethod
-    def _setter_wrapper(obj, member, sig):
-        def invoker(self, value):
-            obj(self, value)
-            message = self._bus.message_new_signal(self._path, "org.freedesktop.DBus.Properties", "PropertiesChanged",
-                                                   'sa{sv}as',
-                                                   self._dbus_interface,
-                                                   { member: { 't': sig, 'v': value } },
-                                                   [])
-            self._bus.send(message, None)
-        return invoker
+    This class works well with these helper classes:
+      - Object.Method
+      - Object.Property
+      - Object.Signal
```

Comment:
I think we should mention that these helper classes take care of getting the Introspection right, so there is no real alternative to using them.  For example, I was almost proposing to ditch `Object.Signal` since `emit_signal` is so easy to use already, but we really should use `Object.Signal` for introspection's sake.